caproto/sync/client.py

```diff
@@ -48,6 +48,7 @@ def recv(circuit):
 def search(pv_name, udp_sock, timeout, *, max_retries=2):
     # Set Broadcaster log level to match our logger.
     b = ca.Broadcaster(our_role=ca.CLIENT)
+    b.our_address = self.udp_sock.getsockname()[:2]
```

Comment:
This should be `udp_sock`, which is passed in to the function signature in L48, now `self.udp_sock`. There is no `self` here because we are inside top-level function, not an instance method.